### DISCUSSION TITLE
Your Ruby version is 2.6.8, but your Gemfile specified 2.7.5

### DISCUSSION THREAD

**Discussion by BraveEvidence** [reactwg/react-native-new-architecture#75] [Q&A] [answered]
I just created a new` react-native` project for version 0.70
`cd` to `iOS` folder and tried running

`bundle install && RCT_NEW_ARCH_ENABLED=1 bundle exec pod install`

I get an error saying 

`Your Ruby version is 2.6.8, but your Gemfile specified 2.7.5`

When I tried to upgrade my ruby version via `home-brew` I get following message

`Warning: ruby 3.1.2_1 already installed`

Here is what I tried

```
brew upgrade
brew update
brew install ruby-build
brew install rbenv
```

```
rbenv install -l
sudo rbenv install 2.7.5
```
`sudo rbenv global 2.7.5`  => I get error saying rbenv: version `2.7.5' not installed

What else should I do to resolve this error

**Comment by fabOnReact** [comment, marked as answer]
I would refer to https://reactnative.dev/docs/environment-setup#cocoapods

inspect your $PATH variable with 

```
echo $PATH
```

and verify what binary are you using when you run commands rbenv or sudo rbenv (`which rbenv` and [rbenv which](https://github.com/rbenv/rbenv#rbenv-which)).

>sudo rbenv global 2.7.5 => I get error saying rbenv: version `2.7.5' not installed

I would not use sudo with rbenv.
Go through the installation and troubleshooting steps in https://github.com/rbenv/rbenv

**Reply by BraveEvidence** [reply]
My setup is correct afaik because I am able to run the app both on `android` and `iOS` without the new architecture

After running` echo $PATH`, below is my output

`/Library/Frameworks/Python.framework/Versions/3.9/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/transformhub/Desktop/flutter/bin :/usr/local/go/bin:/usr/local/share/dotnet:~/.dotnet/tools:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/Postgres.app/Contents/Versions/latest/bin:/Users/transformhub/Library/Android/sdk/platform-tools:/Users/transformhub/Users/transformhub/Desktop/flutter/bin:/Users/transformhub/Android/Sdk/tools:/Users/transformhub/Android/Sdk/tools/bin:/Users/transformhub/Android/Sdk/platform-tools:/Library/Frameworks/Python.framework/Versions/3.9/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/transformhub/Desktop/flutter/bin:/Users/transformhub/.rvm/bin`

Even if I don't use `sudo` for `sudo rbenv global 2.7.5`, I get the same error

**Reply by fabOnReact** [reply]
https://stackoverflow.com/a/19074129/7295772

>`/Users/transformhub/.rvm/bin`

rbenv is not compatible with rvm
https://github.com/rbenv/rbenv#installation

you can also check this and other stackoverflow [discussions](https://stackoverflow.com/questions/10940736/rbenv-not-changing-ruby-version/12150580#12150580)

**Reply by BraveEvidence** [reply]
Thanks but I still don't understand why the issue is 

I added 

```
export PATH="$HOME/.rbenv/bin:$PATH"
eval "$(rbenv init -)"
```

in my zsh

and also did `brew install rbenv ruby-build` which told me I already have

**Comment by BraveEvidence** [comment]
This is how is resolved the issue. First uninstall `rvm` and `rbenv`

Then follow [this](https://github.com/rbenv/ruby-build/issues/1691#issuecomment-983122764) 
Make sure you install the proper ruby version

Then add 

```
export PATH="$HOME/.rbenv/bin:$PATH"
eval "$(rbenv init -)"
```

as the last line in your `.zshrc`

Then run

`env | grep PATH`
`rbenv global 2.7.6`
